### DISCUSSION TITLE
Importing lights from glbs into Spoke

### DISCUSSION THREAD

**Discussion by iatenothingbutriceforthreedays** [mozilla/hubs#2899] [General Questions & Discussion] [answered]
Hey, is there any way to get Spoke to import point light objects from glbs? It seems to just ignore them, so there is no nice way to get lighting information from Blender into Hubs.

**Comment by rawnsley** [comment]
There are two parts to this problem: exporting lights from Blender to GLTF and then importing the lights from the GLTF into Spoke. Do you have example files (blend or glb) that demonstrate the problem?

**Reply by iatenothingbutriceforthreedays** [reply]
Yeah, true. I can export point/sun/spot lights from Blender by selecting the 'Punctual Lights' option when exporting the GLTF (attached an example).
And I can see that at least some of those lights show up when viewing the glb in https://gltf-viewer.donmccurdy.com/.
But none appear in Spoke or in Hubs (after publishing from Spoke).

[light_test.glb.zip](https://github.com/mozilla/hubs/files/5110650/light_test.glb.zip)

**Comment by robertlong** [comment, marked as answer]
Spoke doesn't have support for the KHR_lights_punctual extension. So you can't import lights from glbs into Spoke at this time.

**Reply by iatenothingbutriceforthreedays** [reply]
I solved this for my case by modifying https://github.com/MozillaReality/hubs-blender-exporter slightly to export 'point-light' / 'directional-light' components on Blender light objects. The lights don't render in Spoke but they do show up in Hubs. I might make that into a PR at some point when I have the time.

A simpler solution for a small number of lights would be to manually add custom properties to Blender objects, eg `"point-light":"intensity:1.2;color:#f0f;"`
